src/ZeroCrossImpl.java

```diff
@@ -0,0 +1,177 @@
+
+        return
+
+
+
+                               (zeroCross[0][0] == y && zeroCross[0][1] == y && zeroCross[0][2] == x &&
+                                zeroCross[1][0] == x && zeroCross[1][1] == y && zeroCross[1][2] == y &&
+                                zeroCross[2][0] == y && zeroCross[2][1] == x && zeroCross[2][2] == x) ||
+
+                                      (zeroCross[0][0] == x && zeroCross[1][0] == x && zeroCross[2][0] == y &&
+                                       zeroCross[0][1] == y && zeroCross[1][1] == y && zeroCross[2][1] == x &&
+                                       zeroCross[0][2] == x && zeroCross[1][2] == y && zeroCross[2][2] == y) ||
+
+                               (zeroCross[0][0] == y && zeroCross[0][1] == y && zeroCross[0][2] == x &&
+                                zeroCross[1][0] == x && zeroCross[1][1] == x && zeroCross[1][2] == y &&
+                                zeroCross[2][0] == y && zeroCross[2][1] == x && zeroCross[2][2] == x) ||
+
+                                                       (zeroCross[0][0] == x && zeroCross[1][0] == y && zeroCross[2][0] == x &&
+                                                        zeroCross[0][1] == y && zeroCross[1][1] == x && zeroCross[2][1] == y &&
+                                                        zeroCross[0][2] == y && zeroCross[1][2] == x && zeroCross[2][2] == y) ||
+
+                                (zeroCross[0][0] == y && zeroCross[0][1] == x && zeroCross[0][2] == y &&
+                                 zeroCross[1][0] == x && zeroCross[1][1] == y && zeroCross[1][2] == y &&
+                                 zeroCross[2][0] == x && zeroCross[2][1] == y && zeroCross[2][2] == x) ||
+
+
+                                (zeroCross[0][0] == y && zeroCross[1][0] == x && zeroCross[2][0] == x &&
+                                 zeroCross[0][1] == x && zeroCross[1][1] == y && zeroCross[2][1] == y &&
+                                 zeroCross[0][2] == y && zeroCross[1][2] == x && zeroCross[2][2] == x) ||
+
+                                (zeroCross[0][0] == x && zeroCross[0][1] == y && zeroCross[0][2] == y &&
+                                 zeroCross[1][0] == x && zeroCross[1][1] == y && zeroCross[1][2] == y &&
+                                 zeroCross[2][0] == x && zeroCross[2][1] == y && zeroCross[2][2] == y) ||
+
+
+                                (zeroCross[0][0] == y && zeroCross[1][0] == y && zeroCross[2][0] == x &&
+                                 zeroCross[0][1] == x && zeroCross[1][1] == x && zeroCross[2][1] == y &&
+                                 zeroCross[0][2] == y && zeroCross[1][2] == y && zeroCross[2][2] == x) ||
+
+                                (zeroCross[0][0] == x && zeroCross[0][1] == x && zeroCross[0][2] == y &&
+                                 zeroCross[1][0] == y && zeroCross[1][1] == y && zeroCross[1][2] == x &&
+                                 zeroCross[2][0] == x && zeroCross[2][1] == x && zeroCross[2][2] == y);
+
```

Comment:
That you also can separate to smaller pieces of won parts. And call them one by one.